src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -458,7 +444,7 @@ private static int BinarySearch(string concatenatedNames, ImmutableArray<Node> n
 
         // returns all the symbols in the container corresponding to the node
         private void Bind(
-            int index, INamespaceOrTypeSymbol rootContainer, ArrayBuilder<ISymbol> results, CancellationToken cancellationToken)
+            int index, INamespaceOrTypeSymbol rootContainer, ref TemporaryArray<ISymbol> results, CancellationToken cancellationToken)
```

Comment:
needing to pass by ref seems worrying (unless we have an analyzer to catch this) as i can easily see someone accidentally passing by val and getting broken behavior.

Reply:
The type is marked `NonCopyable`. The analyzer will not let a copy be made intentionally or accidentally.

Reply:
that's awesome!  can you link me to that analyzer?

Reply:
The bulk of the implementation is in https://github.com/dotnet/roslyn-analyzers/blob/master/src/Roslyn.Diagnostics.Analyzers/Core/DoNotCopyValue.cs. Some language-syntax-specific operations are in the derived types.